applications/vila_live/tests/test_tinychat.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import signal
+import subprocess
+import sys
+import time
+import unittest
+
+import requests
+
+
+class TestTinyChat(unittest.TestCase):
+    """Test cases for the TinyChat controller and model worker"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Start the TinyChat controller and model worker before running the tests"""
+        # Start the controller
+        cls.controller_process = subprocess.Popen(
+            ["python3", "-m", "tinychat.serve.controller", "--host", "0.0.0.0", "--port", "10000"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=os.environ,
+            preexec_fn=os.setsid,
+        )
+        time.sleep(10)
+        try:
+            response = requests.get("http://localhost:10000")
+            if response.status_code != 404:
+                cls.tearDownClass()
+                raise Exception("Controller failed to start properly")
+        except requests.exceptions.ConnectionError:
+            cls.tearDownClass()
+            raise Exception("Could not connect to controller")
+
+        # Start the model worker
+        model_path = "/workspace/volumes/models/Llama-3-VILA1.5-8b-AWQ/"
+        quant_path = "/workspace/volumes/models/Llama-3-VILA1.5-8b-AWQ/llm/llama-3-vila1.5-8b-w4-g128-awq-v2.pt"
```

Comment:
these are downloaded within the dockerfile https://github.com/nvidia-holoscan/holohub/blob/3c0b29ad4288ba5a89566d659d98030be353cf44/applications/vila_live/Dockerfile#L68-L69  is it ok to write the absolute path in the unit test?

Reply:
Seems OK to assume fixed paths in the test environment. You could consider adding paths to the test config at `vila_live_testing.yaml` and reading that config file here rather than inlining.